### PR TITLE
[WIP] Major update WIP FOR DISCUSSION

### PR DESCRIPTION
**Pull request by brodybits**
### Platforms affected

All

### What does this PR do?

- Mark new version (major)
- drop Node.js 4 support
- drop Node.js 4, obsolete platforms, and npm@5 workaround no longer needed
- update eslint and other devDependencies (PR #163)
- remove browserify which is under discussion elsewhere (PR #162)
- add package-lock.json, which is needed for coho to work properly as discussed on apache/cordova-coho#174

TODO:
- update tests to use `jasmine` instead of `jasmine-node` as already done in #140

### What testing has been done on this change?

TODO

### Checklist

- ~~[Reported an issue](http://cordova.apache.org/contribute/issues.html) in the JIRA database~~
- ~~Commit message follows the format: "CB-3232: (android) Fix bug with resolving file paths", where CB-xxxx is the JIRA ID & "android" is the platform affected.~~
- ~~Added automated test coverage as appropriate for this change.~~

### CODE REVIEW

**Comment by raphinesse** [comment]
Are there still relevant parts to this PR?

File: package.json [hunk 1]
```diff
@@ -16,6 +16,7 @@
   },
   "scripts": {
     "eslint": "eslint src test pkg",
+
```
Comment (brodybits) [inline package.json]:
```suggestion
```

**Comment by brodybits** [comment]
> Are there still relevant parts to this PR?

Here is what I found after resolving the merge conflicts:

- update devDependencies (duplicate of #163)
- remove specific npm version no longer needed from AppVeyor build
- a quick fix to `src/common/utils.js` to log every alert message if possible, regardless of whether or not it is able to actually show the alert
- commit package-lock.json (I actually do not particularly like doing this any more; I try to use Yarn on other projects and it seems to work much more smoothly)

**Comment by raphinesse** [comment]
I think this PR is obsolete, going to close it for now. Below is my reasoning:

> * update devDependencies (duplicate of #163)

also duplicate of #192 now

> * remove specific npm version no longer needed from AppVeyor build

Fixed in #189

> * a quick fix to `src/common/utils.js` to log every alert message if possible, regardless of whether or not it is able to actually show the alert

I personally don't think we need this, but if you have an argument for it, please raise in separate PR

> * commit package-lock.json (I actually do not particularly like doing this any more; I try to use Yarn on other projects and it seems to work much more smoothly)

Whether we are going to commit it or not, it would have to be recreated from scratch anyway

**Pull request closed without merging**